src/bitcoin-chainstate.cpp

```diff
@@ -0,0 +1,255 @@
+#include <core_io.h>                  // for DecodeHexBlk
+#include <consensus/validation.h>     // for BlockValidationState
+#include <chainparams.h>              // for Params, SelectParams, CChainParams
+#include <node/blockstorage.h>        // for fReindex
+#include <util/system.h>              // for gArgs, ArgsManager
+#include <util/thread.h>              // for TraceThread
+#include <scheduler.h>                // for CScheduler
+#include <script/sigcache.h>          // for InitSignatureCache
+
+using node::fReindex;
+using node::LoadChainstate;
+using node::VerifyLoadedChainstate;
+
+const extern std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
+
+// Adapted from rpc/mining.cpp
+class submitblock_StateCatcher final : public CValidationInterface
+{
+public:
+    uint256 hash;
+    bool found;
+    BlockValidationState state;
+
+    explicit submitblock_StateCatcher(const uint256 &hashIn) : hash(hashIn), found(false), state() {}
+
+protected:
+    void BlockChecked(const CBlock& block, const BlockValidationState& stateIn) override {
+        if (block.GetHash() != hash)
+            return;
+        found = true;
+        state = stateIn;
+    }
+};
+
+
+int main(int argc, char* argv[]) {
+    assert(argc == 2);
+    std::filesystem::path abs_datadir = std::filesystem::absolute(argv[1]);
+    std::filesystem::create_directories(abs_datadir);
+    gArgs.ForceSetArg("-datadir", abs_datadir.string());
+
+    SelectParams(CBaseChainParams::MAIN);
+    const CChainParams& chainparams = Params();
+
+    init::SetGlobals();  // ECC_Start, etc.
+
+    // Initialize signatureCache cuz it's used by...
+    //     <- VerifyECDSASignature
+    //     <- CheckECDSASignature
+    //     <- EvalChecksigPreTapscript
+    //     <- EvalScript
+    //     <- VerifyScript
+    //     <- CScriptCheck::()
+    //     <- CheckInputScripts
+    //     <- ConnectBlock
+    //     <- ConnectTip
+    //     <- ActivateBestChainStep
+    //     <- ActivateBestChain
+    //     <- ProcessNewBlock
+    InitSignatureCache();
+
+    // Initialize g_scriptExecutionCache{,Hasher} cuz it's used by...
+    //     <- CheckInputScripts
+    //     <- ConnectBlock
+    //     <- ConnectTip
+    //     <- ActivateBestChainStep
+    //     <- ActivateBestChain
+    //     <- ProcessNewBlock
+    InitScriptExecutionCache();
+
+    // START scheduler for RegisterSharedValidationInterface
+    CScheduler scheduler{};
+    // Start the lightweight task scheduler thread
+    scheduler.m_service_thread = std::thread(util::TraceThread, "scheduler", [&] { scheduler.serviceQueue(); });
+
+    // Gather some entropy once per minute.
+    scheduler.scheduleEvery([]{
+        RandAddPeriodic();
+    }, std::chrono::minutes{1});
+    // END scheduler for RegisterSharedValidationInterface
+
+    GetMainSignals().RegisterBackgroundSignalScheduler(scheduler);
+
+    ChainstateManager chainman;
+
+    auto rv = LoadChainstate(false,
```

Comment:
In commit "build: Add example bitcoin-chainstate executable" (01840e6161396ed0ab30a122921f67b8dce45499)

The LoadChainstate and VerifyLoadedChainstate calls here would be cleaner with  b7c7f64dc86f0fcdf07cb1e765f7cccc3a3c8897 suggested in https://github.com/bitcoin/bitcoin/pull/23280#pullrequestreview-825684290 and would let this code print specific error messages instead failing generically and discarding the return codes.